instrumentation/vertx/vertx-sql-client-4.0/javaagent/src/test/java/io/opentelemetry/javaagent/instrumentation/vertx/v4_0/sql/VertxSqlClientTest.java

```diff
@@ -0,0 +1,251 @@
+        .toCompletableFuture()
+        .get(30, TimeUnit.SECONDS);
+  }
+
+  @AfterAll
+  static void cleanUp() {
+    pool.close();
+    vertx.close();
+    container.stop();
+  }
+
+  @Test
+  void testSimpleSelect() throws Exception {
+    CompletableFuture<Object> result = new CompletableFuture<>();
+    result.whenComplete((rows, throwable) -> testing.runWithSpan("callback", () -> {}));
+    testing.runWithSpan(
+        "parent",
+        () ->
+            pool.query("select * from test")
+                .execute(
+                    rowSetAsyncResult -> {
+                      if (rowSetAsyncResult.succeeded()) {
+                        result.complete(rowSetAsyncResult.result());
+                      } else {
+                        result.completeExceptionally(rowSetAsyncResult.cause());
+                      }
+                    }));
+    result.get(30, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL),
+                span ->
+                    span.hasName("SELECT tempdb.test")
+                        .hasKind(SpanKind.CLIENT)
+                        .hasParent(trace.getSpan(0))
+                        .hasAttributesSatisfyingExactly(
+                            equalTo(DB_NAME, DB),
+                            equalTo(DB_USER, USER_DB),
+                            equalTo(DB_STATEMENT, "select * from test"),
+                            equalTo(DB_OPERATION, "SELECT"),
+                            equalTo(DB_SQL_TABLE, "test"),
+                            equalTo(NET_PEER_NAME, "localhost"),
+                            equalTo(NET_PEER_PORT, port)),
+                span ->
+                    span.hasName("callback")
+                        .hasKind(SpanKind.INTERNAL)
+                        .hasParent(trace.getSpan(0))));
+  }
+
+  @Test
+  void testInvalidQuery() throws Exception {
+    CountDownLatch latch = new CountDownLatch(1);
+    CompletableFuture<Object> result = new CompletableFuture<>();
+    result.whenComplete(
+        (rows, throwable) -> testing.runWithSpan("callback", () -> latch.countDown()));
+    testing.runWithSpan(
+        "parent",
+        () ->
+            pool.query("invalid")
+                .execute(
+                    rowSetAsyncResult -> {
+                      if (rowSetAsyncResult.succeeded()) {
+                        result.complete(rowSetAsyncResult.result());
+                      } else {
+                        result.completeExceptionally(rowSetAsyncResult.cause());
+                      }
+                    }));
+
+    latch.await(30, TimeUnit.SECONDS);
+
+    testing.waitAndAssertTraces(
+        trace ->
+            trace.hasSpansSatisfyingExactly(
+                span -> span.hasName("parent").hasKind(SpanKind.INTERNAL),
+                span ->
+                    span.hasName("tempdb")
```

Comment:
[Spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/database.md) says "When it's otherwise impossible to get any meaningful span name, `db.name` or the tech-specific database name MAY be used." Because the query is invalid it is not possible to extract anything from it.

Reply:
Cool, thanks!